[2025-06-01T17:55:39.682+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: movie_recommendation_pipeline.train_hybrid_model manual__2025-06-01T17:55:13.215545+00:00 [queued]>
[2025-06-01T17:55:39.690+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: movie_recommendation_pipeline.train_hybrid_model manual__2025-06-01T17:55:13.215545+00:00 [queued]>
[2025-06-01T17:55:39.691+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 3
[2025-06-01T17:55:39.701+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): train_hybrid_model> on 2025-06-01 17:55:13.215545+00:00
[2025-06-01T17:55:39.708+0000] {standard_task_runner.py:60} INFO - Started process 309 to run task
[2025-06-01T17:55:39.710+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'movie_recommendation_pipeline', 'train_hybrid_model', 'manual__2025-06-01T17:55:13.215545+00:00', '--job-id', '1195', '--raw', '--subdir', 'DAGS_FOLDER/mlops_movie_dag.py', '--cfg-path', '/tmp/tmp_g17b_cz']
[2025-06-01T17:55:39.711+0000] {standard_task_runner.py:88} INFO - Job 1195: Subtask train_hybrid_model
[2025-06-01T17:55:39.748+0000] {task_command.py:423} INFO - Running <TaskInstance: movie_recommendation_pipeline.train_hybrid_model manual__2025-06-01T17:55:13.215545+00:00 [running]> on host 3699593ac5c8
[2025-06-01T17:55:39.807+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='movie_recommendation_pipeline' AIRFLOW_CTX_TASK_ID='train_hybrid_model' AIRFLOW_CTX_EXECUTION_DATE='2025-06-01T17:55:13.215545+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-01T17:55:13.215545+00:00'
[2025-06-01T18:00:42.906+0000] {mlops_movie_dag.py:13} INFO -                       ❗❗❗ AUTHORIZATION REQUIRED ❗❗❗                      


Open the following link in your browser to authorize the client:
https://dagshub.com/login/oauth/authorize?state=98b4da89-0883-4d09-8f1f-3a7af93dc80d&client_id=32b60ba385aa7cecf24046d8195a71c07dd345d9657977863b52e7748e0f0f28&middleman_request_id=3f43623cc3197e78f5217fb402f0ebb9b357c6479afd16988704165fd8ae192c


Traceback (most recent call last):
  File "/opt/***/src/models/train_hybrid_model.py", line 17, in <module>
    dagshub.init(repo_owner='sacer11', repo_name='MLOps_movie_rec_apr25', mlflow=True)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/common/init.py", line 86, in init
    repo_api = RepoAPI(f"{repo_owner}/{repo_name}", host=host)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/common/api/repo.py", line 78, in __init__
    self.auth = dagshub.auth.get_authenticator(host=host)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/auth/tokens.py", line 374, in get_authenticator
    return _get_token_storage(**kwargs).get_authenticator(**kwargs)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/auth/tokens.py", line 122, in get_authenticator
    token = self.get_token_object(host, fail_if_no_token, **kwargs)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/auth/tokens.py", line 187, in get_token_object
    good_token = oauth.oauth_flow(host, **kwargs)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/auth/oauth.py", line 46, in oauth_flow
    token = oauth_flow_post(flow_data)
  File "/home/***/.local/lib/python3.10/site-packages/dagshub/auth/oauth.py", line 117, in oauth_flow_post
    raise Exception(f"Error while getting OAuth code: HTTP {res.status_code}, Body: {res.json()}")
  File "/home/***/.local/lib/python3.10/site-packages/httpx/_models.py", line 755, in json
    return jsonlib.loads(self.content.decode(encoding), **kwargs)
  File "/usr/local/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

[2025-06-01T18:00:42.907+0000] {python.py:201} INFO - Done. Returned value was: None
[2025-06-01T18:00:42.915+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=movie_recommendation_pipeline, task_id=train_hybrid_model, execution_date=20250601T175513, start_date=20250601T175539, end_date=20250601T180042
[2025-06-01T18:00:42.959+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2025-06-01T18:00:42.977+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
